core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java

```diff
@@ -0,0 +1,300 @@
+    if (activeFilterCount == 0) {
+      return String.format(Messages.CapellaSearchResult_Label, queryLabel, totalOccurrenceCount, matchedElementsCount,
+          matchedProjectsCount);
+    }
+    int displayedOccurrenceCount = 0; //getDisplayedOccurrenceCount();
+    return String.format(Messages.CapellaSearchResult_Label_With_Active_Filters, queryLabel, totalOccurrenceCount,
+        matchedElementsCount, matchedProjectsCount, totalOccurrenceCount - displayedOccurrenceCount, activeFilterCount);
+  }
+
+  @Override
+  public String getTooltip() {
+    return getLabel();
+  }
+
+  @Override
+  public ImageDescriptor getImageDescriptor() {
+    return Activator.getDefault().getImageDescriptor("search.gif"); //$NON-NLS-1$
+  }
+
+  @Override
+  public CapellaSearchQuery getQuery() {
+    return capellaSearchQuery;
+  }
+
+  @Override
+  public IEditorMatchAdapter getEditorMatchAdapter() {
+    return null;
+  }
+
+  @Override
+  public IFileMatchAdapter getFileMatchAdapter() {
+    return null;
+  }
+
+  /**
+   * Get all projects from matches
+   * 
+   * @return
+   */
+  public Set<Object> getProjects() {
+    return searchEntries.keySet();
+  }
+
+  @Override
+  public MatchFilter[] getAllMatchFilters() {
+    return new MatchFilter[] { //
+//        CapellaSearchMatchFilter.NOT_MODIFIABLE, //
+//        CapellaSearchMatchFilter.CAPELLA_ELEMENT, //
+//        CapellaSearchMatchFilter.REPRESENTATION //
+    };
+  }
+
+  /////////
+  
+  
+  /**
+   * {@inheritDoc}
+   */
+  public AbstractCapellaSearchEntry insert(Object file, AbstractCapellaSearchEntry entry, boolean notify) {
+    if (searchEntries.get(file) == null) {
+      searchEntries.put(file, new ArrayList<Object>());
+    }
+    insert2(searchEntries.get(file), entry, notify);
+    return entry;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public AbstractCapellaSearchEntry insert(Object file, AbstractCapellaSearchEntry entry, Object eTypedElem, String valuation,
+      boolean notify) {
+    return insert5(entry, eTypedElem, valuation, notify, file);
+  }
+
+  /**
+   * Inserts & merges an entry sequence into an existing entry sequence hierarchy
+   * 
+   * @param currentEntrySubHierarchyCollection
+   * @param entryToInsert
+   * @param notify
+   */
+  private void insert2(Collection<Object> currentEntrySubHierarchyCollection, AbstractCapellaSearchEntry entryToInsert,
+      boolean notify) {
+
+      boolean alreadyExist = false;
+      for (Object currentEntrySubHierarchy : currentEntrySubHierarchyCollection) {
+        if (alreadyExist = currentEntrySubHierarchy.equals(entryToInsert)) {
+          updateInsertionPoint((AbstractCapellaSearchEntry) currentEntrySubHierarchy, (AbstractCapellaSearchEntry) entryToInsert);
+          insert3(currentEntrySubHierarchy, entryToInsert.getChildren(), notify);
+          break;
+        }
+      }
+      if (!alreadyExist) {
+        currentEntrySubHierarchyCollection.add(entryToInsert);
+        if (notify) {

```

Comment:
This has not been finished, isn't it?

Reply:
I didn't use the notify, will be removed.